src/_guides/libraries/objective-c-interop.md

```diff
@@ -0,0 +1,224 @@
+ignore rules at the top of the generated file. You can find a full list of
+config options in the [ffigen readme]({{page.ffigen}}#configurations).
+
+### Generating the bindings
+
+To generate the bindings, navigate to the example directory, and run
+`dart run ffigen`. This will search in the pubspec.yaml file for an `ffigen:`
+config. If your config is in a standalone file, then use
+`dart run ffigen --config foo.yaml`.
+
+For our example, this will generate
+[avf_audio_bindings.dart]({{page.example}}/avf_audio_bindings.dart).
+
+This file contains a class called `AVFAudio`, which is the native library
+wrapper that holds all the loaded function pointers etc. The other classes
+in this file are all Dart wrappers around the Objective-C interfaces that
+we need, such as `AVAudioPlayer` and its dependencies.
+
+### Using the bindings
+
+Now we're ready to load and interact with our library. Our example app,
+[play_audio.dart]({{page.example}}/play_audio.dart), loads and plays
+audio files passed as command line arguments.
+The first step is to load the dylib and instantiate the native
+library, `AVFAudio`.
+
+```dart
+import 'dart:ffi';
+import 'avf_audio_bindings.dart';
+
+const _dylibPath =
+    '/System/Library/Frameworks/AVFAudio.framework/Versions/Current/AVFAudio';
+
+void main(List<String> args) async {
+  final lib = AVFAudio(DynamicLibrary.open(_dylibPath));
+```
+
+Since we're loading an internal library, we're pointing at an internal
+framework dylib. You can also load your own `foo.dylib` file, or if the
+library is statically linked into your app (often the case on iOS) you
+can use `DynamicLibrary.process()`:
+
+```dart
+  final lib = AVFAudio(DynamicLibrary.process());
+```
+
+We want to play each of the files in the command line args one by one.
+For each arg, the we first have to convert the Dart `String` to an
+Objective-C `NSString`. The generated `NSString` wrapper has a convenient
+constructor that handles this conversion, and a `toString()` method that
+converts it back to a Dart `String`.
+
+```dart
+  for (final file in args) {
+    final fileStr = NSString(lib, file);
+    print('Loading $fileStr');
+```
+
+The audio player expects an `NSURL`, so next we use the [`fileURLWithPath:`](
+{{page.appledoc}}/foundation/nsurl/1410828-fileurlwithpath?language=objc)
+method to convert the `NSString` to an `NSURL`. Since `:` is not a
+valid character in a Dart method name, it has been translated to `_`.
+
+```dart
+    final fileUrl = NSURL.fileURLWithPath_(lib, fileStr);
+```
+
+Now we can construct our `AVAudioPlayer`. Constructing an Objective-C
+object has two stages. `alloc` allocates the memory for the object, but
+doesn't initialize it. Methods with names starting with `init*` do the
+initialization. Some interfaces also provide `new*` methods that do
+both these steps.
+
+The init method we will use is [`initWithContentsOfURL:error:`](
+{{page.appledoc}}/avfaudio/avaudioplayer/1387281-initwithcontentsofurl?language=objc).
```

Comment:
Maybe we can reconfiguring this sentence to be more like what a user following along can do:

```suggestion
To then initialize the `AVAudioPlayer`,
use the [`initWithContentsOfURL:error:`][] method.

[`initWithContentsOfURL:error:`]: {{page.appledoc}}/avfaudio/avaudioplayer/1387281-initwithcontentsofurl?language=objc
```